README.md
# 🚀🚀 Gundam Evolution Challenges 🚀🚀
> If any challenge types, units, or unit groups are missing please create an issue so I can add them in.

This simple application is a stand-in for Gundam Evolution's challenges system which, as of right now, is only accessible between matches. This makes it hard to know which challenges you have left and for which units. Now with this app you can just open a second monitor or alt tab and keep track of the challenges you would like to work on. All of your data is stored locally to your computer so you do not need to create an account or anything like that.

## Usage
Using this app is meant to be as simple as possible.
1. Clone the repo or navigate to the [hosted URL](https://gundam-evo.gmurray.dev/)
2. Add your daily and weekly challenges
    * Select the challenge type (kills, wins, etc.), amount (# of kills etc.), and the group of units. You can add additional unit groups if necessary. Please let me know if there are any missing and I will add them ASAP.
3. Update your progress on the app as you make progress in the game
4. Reset your dailies/weeklies whenever new ones arrive for you to work on

## Potential Features
I consider this app to be mostly feature-complete because I want to keep it as simple and easy to use as possible. That being said I could still add a few things:

- [X] Add timers based on daily/weekly reset
- [X] Better icons
- [X] Delete challenges
- [ ] Online data persistence


src/CHANGELOG.md
## [2.7.0] - 2023-05-31
### Added
- Right click challenges to edit, remove, or rearrange
---
## [2.6.2] - 2023-05-03
### Added
- Unit grouping for Pale Rider, Gundam, Turn A, Dynames
---
## [2.6.1] - 2023-04-16
### Changed
- Fixed and improved UX 
---
## [2.6.0] - 2023-04-13
### Added
- User preferences!
  - hide completed challenges
  - ✨themes✨
---
## [2.5.0] - 2023-04-13
### Added
- Unit grouping for GM, Sazabi, Dom Trooper, Dynames
- Unit grouping for Asshimar, Zaku II Ranged, Exia, Dynames
- Undo feature when to undo a recently deleted challenge progress
---
## [2.4.2] - 2023-04-04
### Added
- Unit grouping for Pale Rider, Gundam, Sazabi, Marasai
---
## [2.4.1] - 2023-03-28
### Added
- Unit grouping for Pale Rider, Turn A, GM Sniper, Nu Gundam
---
## [2.4.0] - 2023-03-23
### Changed
- Recommended units are all shown now rather than just the top 3. The first shown and then you can hover/tap to see the full list.
- Summary UX improvements - smaller elements for each unit with more readable progress values
### Added
- Summary page list can now be sorted by name and number of challenges
---
## [2.3.5] - 2023-03-23
### Added
- Increment *and* decrement challenge progress
---
## [2.3.4] - 2023-03-22
### Changed
- Recommended unit colors are now gold/silver/bronze
---
## [2.3.3] - 2023-03-20
### Added
- Added Pale Rider, GM, Turn A, Hyperion Gundam grouping
---
## [2.3.2] - 2023-03-20
### Fixed
- Summary units would not always accurately represent all units with challenges
---
## [2.3.1] - 2023-03-20
### Added
- Feedback form
---
## [2.3.0] - 2023-03-20
### Added
- 🚨SUMMARY SECTION🚨 This section shows your challenges by unit. Simply set up your challenges on the "List" tab then switch to the "Summary" tab to view your progress for each mobile suit!
### Changed
- UX improvements on list tab
---
## [2.2.0] - 2023-03-18
### Added
- Unit recommendations based off how many challenges each unit appears in
- Increment damage value by 5,000 instead of 1,000 when total damage goal is 100,000
### Fixed
- Sometimes challenges wouldn't save properly after selecting type, total, and units
---
## [2.1.2] - 2023-03-18
### Fixed
- Clicking "Changes" when notified of a new app version works as expected
---
## [2.1.1] - 2023-03-18
### Added
- Added Sazabi, Asshimar, Dom Trooper, Heavyarms unit grouping
---
## [2.1.0] - 2023-03-18
### Changed
- Selecting challenge type, total, and units automatically adds the challenge to your list (previously you would have to click a button to save the challenge)
---
## [2.0.0] - 2023-03-18
### Added
- This changelog, which can be accessed via the new update notification and in the app page footer

src/data/units.ts
export type Unit = {
  id: string;
  name: string;
  imageName: string;
  websiteId: number;
  description: string;
  model: string;
};

export const unitOptions: Record<Unit['id'], Unit> = {
  pale_rider: {
    id: 'pale_rider',
    name: 'Pale Rider',
    imageName: 'pale_rider.png',
    websiteId: 1,
    description: `A prototype mobile suit developed by the Earth Federation Forces. Its unique HADES operating system grants superior combat performance, but puts an increased strain on the pilot and has limited operation time.`,
    model: `RX-80PR`,
  },
  gundam: {
    id: 'gundam',
    name: 'Gundam',
    imageName: 'gundam.png',
    websiteId: 2,
    description: `Trailing the Principality of Zeon in Mobile Suit development, the Federation developed this machine as part of Operation V to turn the tide. Boasts remarkable performance. Piloted by Amuro Ray.`,
    model: `RX-78-2`,
  },
  zaku_ranged: {
    id: 'zaku_ranged',
    name: 'Zaku II (Ranged)',
    imageName: 'zaku_ranged.png',
    websiteId: 3,
    description: `A mass-produced mobile suit developed by the Principality of Zeon. After overwhelming the Federation during the early days of the One Year War, the Zaku's versatile design spawned numerous variations. This model has an anti-mobile-suit Zaku Machine Gun as its main weapon.`,
    model: `MS-06`,
  },
  barbatos: {
    id: 'barbatos',
    name: 'Gundam Barbatos',
    imageName: 'barbatos.png',
    websiteId: 4,
    description: `One of the 72 Gundam Frames developed during the Calamity War. First unearthed on Mars, it eventually became Tekkadan's dominant battlefield weapon. Piloted by Mikazuki Augus.`,
    model: `ASW-G-08`,
  },
  sazabi: {
    id: 'sazabi',
    name: 'Sazabi',
    imageName: 'sazabi.png',
    websiteId: 5,
    description: `A Mobile Suit for Newtype pilots developed by Neo Zeon. Its cutting-edge Psycho-Frame provides unparalleled responsiveness, making the Sazabi the pinnacle of Zeon's Newtype design tech. Piloted by Char Aznable.`,
    model: `MSN-04`,
  },
  methuss: {
    id: 'methuss',
    name: 'Methuss',
    imageName: 'methuss.png',
    websiteId: 6,
    description: `The Methuss is the first transforming attack mobile suit developed by the AEUG. Easy to handle thanks to its simple transformation mechanism, it can also play support thanks to its energy supply cable.`,
    model: `MSA-005`,
  },
  gm_sniper: {
    id: 'gm_sniper',
    name: 'GM Sniper II',
    imageName: 'gm_sniper.png',
    websiteId: 7,
    description: `An offshoot of the GM, developed by the Federation Forces during the last days of the One Year War. With specs on par with the RX-78 Gundam and a sophisticated sensor array in its head, the GM Sniper II can use its head visor to snipe targets across staggeringly long distances.`,
    model: `RGM-79SP`,
  },
  asshimar: {
    id: 'asshimar',
    name: 'Asshimar',
    imageName: 'asshimar.png',
    websiteId: 8,
    description: `The first transforming mobile armor capable of self-sustaining flight within the Earth's atmosphere, the Asshimar's lifting body structure allows it to alter its aerodynamics by transforming its upper body into a disc shape.`,
    model: `NRX-044`,
  },
  dom_trooper: {
    id: 'dom_trooper',
    name: 'DOM Trooper',
    imageName: 'dom_trooper.png',
    websiteId: 9,
    description: `Although the DOM Trooper was originally intended to serve as ZAFT's next-generation machine, the Clyne Faction actually completed its design and deployed it in battle. The DOM Trooper's large, leg-mounted hovering units provide swift ground movement across all manner of terrain.`,
    model: `ZGMF-XX09T`,
  },
  turn_a: {
    id: 'turn_a',
    name: 'Turn A Gundam',
    imageName: 'turn_a.png',
    websiteId: 10,
    description: `A mysterious mobile suit found within a statue known as the White Doll. It was built with extraordinary technology, and legend speaks of it laying waste to Earth's civilization in a climactic battle long ago. Piloted by Loran Cehack.`,
    model: `WD-M01`,
  },
  guntank: {
    id: 'guntank',
    name: 'Guntank',
    imageName: 'guntank.png',
    websiteId: 11,
    description: `The first mobile suit successfully developed by the Earth Federation Forces as part of Operation V. Its tank-like design reveals a disparity between Federation and Zeon tech. It can launch the transforming Core Fighter stored inside its body.`,
    model: `RX-75`,
  },
  gm: {
    id: 'gm',
    name: 'GM',
    imageName: 'gm.png',
    websiteId: 12,
    description: `The main mobile suit of the Federation Forces. A low-cost version of the RX-78, GMs were mass produced and frequently deployed in combat alongside Ball combat pods.`,
    model: `RGM-79`,
  },
  marasai: {
    id: 'marasai',
    name: 'Marasai',
    imageName: 'marasai.png',
    websiteId: 14,
    description: `Developed by Anaheim Electronics, the Marasai was deployed by the Titans during the Gryps Conflict, and was later used by Neo Zeon Remnants.`,
    model: `RMS-108`,
  },
  exia: {
    id: 'exia',
    name: 'Gundam Exia',
    imageName: 'exia.png',
    websiteId: 13,
    description: `One of the four Gundams used by Celestial Being to end war through armed intervention. Developed to excel in melee combat under any conditions. Piloted by Setsuna F. Seiei.`,
    model: `GN-001`,
  },
  mahiroo: {
    id: 'mahiroo',
    name: 'Mahiroo',
    imageName: 'mahiroo.png',
    websiteId: 15,
    description: `The main mobile suit of the lunar Ghingnham Fleet. With a head full of sensors and an internal structure lined with artificial muscle fibers, the Mahiroo is clearly cut from a different cloth than other mobile suits of its time period.`,
    model: `G-838`,
  },
  zaku_melee: {
    id: 'zaku_melee',
    name: 'Zaku II (Melee)',
    imageName: 'zaku_melee.png',
    websiteId: 16,
    description: `A mass-produced mobile suit developed by the Principality of Zeon, the Zaku's versatile design spawned numerous variations. Vice Admiral Dozle Zabi's personal machine of this model is adorned with engravings and equipped with a large Heat Hawk.`,
    model: `MS-06`,
  },
  unicorn: {
    id: 'unicorn',
    name: 'Unicorn Gundam',
    imageName: 'unicorn.png',
    websiteId: 17,
    description:
      'The Unicorn utilizes the NT-D System, which activates against Newtypes and Cyber-Newtypes. Piloted by Banagher Links, whose psycho-waves resonate with the NT-D to become a supernatural force.',
    model: `RX-0`,
  },
  nu: {
    id: 'nu',
    name: 'Nu Gundam',
    imageName: 'nu.png',
    websiteId: 18,
    description:
      'The Nu Gundam was designed specifically for Newtypes, whose basic design came from the Federation’s ace Amuro Ray. The Psycho-Frame provides unparalleled responsiveness, and is equipped with Fin Funnels, guided weapons that are deployed to provide beam attacks and shield defenses at the same time.',
    model: `RX-93`,
  },
  hyperion: {
    id: 'hyperion',
    name: 'Hyperion Gundam',
    imageName: 'hyperion.png',
    websiteId: 19,
    description:
      'A unit that possesses two defensive arms—a shield and a barrier.While balanced for attack or defense for frontline and area control, its Armure Lumiere Lancer can break through enemy lines.',
    model: `CAT1-X1/3`,
  },
  heavyarms: {
    id: 'heavyarms',
    name: 'Gundam Heavyarms Custom [EW]',
    imageName: 'heavyarms.png',
    websiteId: 20,
    description:
      'A unit with superb offensive capabilities, featuring a main weapon that does not require reloading.It can destroy individual enemies with its Double Gatling Gun (Dual Wield) and suppress an area with its Hatch Full Open (Barrage).',
    model: 'XXXG-01H2',
  },
  dynames: {
    id: 'dynames',
    name: 'Gundam Dynames',
    imageName: 'dynames.png',
    websiteId: 21,
    description: `The unit's ability to swap its main weapons makes it effective at any range.In TRANS-AM mode, the GN Sniper Rifle deals increased damage.`,
    model: 'GN-002'
  }
};
